sharkfuser/benchmarks/driver.cpp

```diff
@@ -165,7 +165,227 @@ benchmarkConvFprop(int64_t n, int64_t c, int64_t d, int64_t h, int64_t w,
+  auto wDims = (s == 2) ? std::vector<int64_t>{k, fc, y, x}
+                        : std::vector<int64_t>{k, fc, z, y, x};
+  auto convStride =
+      (s == 2) ? std::vector<int64_t>{u, v} : std::vector<int64_t>{t, u, v};
+  auto convPadding =
+      (s == 2) ? std::vector<int64_t>{p, q} : std::vector<int64_t>{o, p, q};
+  auto convDilation =
+      (s == 2) ? std::vector<int64_t>{l, j} : std::vector<int64_t>{m, l, j};
+
+  // Calculate output dimensions (DY shape) using the same inference as forward
+  auto dyDims = getConvInferredOutputShape(xDims, wDims, convDilation,
+                                           convPadding, convStride);
+
+  auto xStride =
+      (imageLayout == "NCHW" || imageLayout == "NCDHW")
+          ? generateStrideFromDim(xDims, getContiguousStrideOrder(xDims.size()))
+          : generateStrideFromDim(xDims,
+                                  getChannelsLastStrideOrder(xDims.size()));
+  auto dyStride = (outputLayout == "NCHW" || outputLayout == "NCDHW")
+                      ? generateStrideFromDim(
+                            dyDims, getContiguousStrideOrder(dyDims.size()))
+                      : generateStrideFromDim(
+                            dyDims, getChannelsLastStrideOrder(dyDims.size()));
+  auto wStride =
+      (filterLayout == "NCHW" || filterLayout == "NCDHW")
+          ? generateStrideFromDim(wDims, getContiguousStrideOrder(wDims.size()))
+          : generateStrideFromDim(wDims,
+                                  getChannelsLastStrideOrder(wDims.size()));
+
+  // Build graph for the given handle (device), validate and compile it.
+  Graph graph;
+
+  // Set unique name to prevent concurrent invocations from polluting cache.
+  auto graphName =
+      std::format("benchmark_conv_wgrad_n{}_c{}_d{}_h{}_w{}_g{}_k{"
+                  "}_z{}_y{}_x{}_t{}_u{}_v{}_o{}"
+                  "_p{}_q{}_m{}_l{}_j{}_S{}_I{}_O{}_F{}",
+                  n, c, d, h, w, g, k, z, y, x, t, u, v, o, p, q, m, l, j, s,
+                  imageLayout, outputLayout, filterLayout);
+  graph.setName(graphName);
+
+  graph.setIODataType(DataType::Float)
+      .setComputeDataType(DataType::Float)
+      .setIntermediateDataType(DataType::Float);
+
+  auto dyT = graph.tensor(
+      TensorAttr().setName("dy").setDim(dyDims).setStride(dyStride).setDataType(
+          convIOType));
+
+  auto xT = graph.tensor(TensorAttr()
+                             .setName("input")
+                             .setDim(xDims)
+                             .setStride(xStride)
+                             .setDataType(convIOType));
+
+  auto convAttr = ConvWGradAttr()
+                      .setStride(convStride)
+                      .setPadding(convPadding)
+                      .setDilation(convDilation)
+                      .setName("conv_wgrad");
+
+  auto dwT = graph.convWGrad(dyT, xT, convAttr);
+  dwT->setDim(wDims).setStride(wStride).setOutput(true).setDataType(convIOType);
```

Comment:
Can we drop the `.setStride(wStride)` here, since it is inferred when unspecified - should help catch issues in the inference logic if things don't match up.